### ISSUE TITLE
[Feature Request] Support HTTP/2 requests

### ISSUE THREAD

**Issue by aldaris** [Kong/insomnia#997]
- Insomnia Version: 5.16.6 (5.16.6.2073)
- Operating System: OSX

## Details

Currently when Insomnia makes its requests to an HTTPS enabled site, it only uses HTTP/1.1. It would be great if Insomnia would utilize HTTP/2 requests via ALPN.

**Comment by gschier** [comment]
This _should_ be easy to add, except the `node-libcurl` project doesn't look like it is built with HTTP/2 compatibility when targeting Windows. See https://github.com/peters/curl-for-windows

I still think this will be good to have eventually 👍

**Comment by WingGithub** [comment]
I hope someone is working on this. http2 adoption is getting pretty high and this is a blocking issue for us. I have to disable http2 in our development server and I can't recommend this tool to devOps if it won't work on production.

**Comment by gschier** [comment]
Just a note: Support for HTTP/2 is already included in the OSX and Linux versions of Insomnia, since they are linked with the `nghttp2` library. 

I think the only thing that needs to be done is to allow the user to supply the [CURLOPT_HTTP_VERSION](https://curl.haxx.se/libcurl/c/CURLOPT_HTTP_VERSION.html) option

**Comment by FallingSnow** [comment]
@gschier HTTP/2 does not appear to be working for me.

```sh
$ uname -a
Linux laptop 5.0.5-arch1-1-ARCH #1 SMP PREEMPT Wed Mar 27 17:53:10 UTC 2019 x86_64 GNU/Linux
```

When using the firefox extension [RESTClient](https://addons.mozilla.org/en-US/firefox/addon/restclient/) it works as expected.
![Screenshot from 2019-04-01 16-08-33](https://user-images.githubusercontent.com/2512197/55365194-6f43fe80-5498-11e9-875f-bda875d13b0e.png)

However Insomnia is another story. It reacts the same way as Postman.

![Screenshot from 2019-04-01 16-08-18](https://user-images.githubusercontent.com/2512197/55365195-70752b80-5498-11e9-96de-78ffc8e7a07c.png)

![Screenshot from 2019-04-01 16-10-12](https://user-images.githubusercontent.com/2512197/55365243-a9ad9b80-5498-11e9-801a-3c4a9dd0c6ee.png)

**Comment by WingGithub** [comment]
@FallingSnow , there may be some other issue in your use case. Postman works with HTTP2 and the error message I get when I try to use insomnia with http2 looks like this: Error: SSL connect error.

**Comment by FallingSnow** [comment]
Correct, postman also has an [issue open](https://github.com/postmanlabs/postman-app-support/issues/2701) for using http2.

It looks like the error you're receiving with Insomnia has to do with SSL rather than HTTP2.

**Comment by FallingSnow** [comment]
Good news. The develop branch does in fact support HTTP2 on at least Linux. I just followed the develop instructions to compile and start the app and bingo, it worked!

**Comment by alexblack** [comment]
Any update on this? Does Insomnia support HTTP2 now on OSX?

**Comment by crystalfp** [comment]
Insomnia is based on libcurl on Windows, correct? So it should be able to access HTTP/2 servers as I do with curl that sets `--http2` by default:
```
"C:\apps\curl\bin\curl.exe" \
--cacert "D:/server/cert/localhost-cert.pem" \
-H "Content-Type: application/json" \
https://localhost:3111/api/project
```
Hope it helps
mario

**Comment by develohpanda** [comment]
Looks like the issue was left open, but **Insomnia does support HTTP/2 now** 🎉 🎉 merged with https://github.com/Kong/insomnia/pull/2276, released in [2020.3.3](https://insomnia.rest/changelog/core/2020.3.0).

![image](https://user-images.githubusercontent.com/4312346/88971782-a0407780-d308-11ea-8818-10ee9a7ce010.png)
